tk 0.5.1 not C23 ready

Trying to compile rubygem-tk 0.5.1 with development stage gcc15 (which defaults to `-std=gnu23`) causes the following error:

```
gcc -I. -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I. -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I. -I/usr/include -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I. -I/usr/include -DHAVE_RUBY_NATIVE_THREAD_P -DHAVE_RB_ERRINFO -DHAVE_RB_HASH_LOOKUP -DHAVE_RB_PROC_NEW -DHAVE_RB_SOURCEFILE -DHAVE_RB_THREAD_CHECK_TRAP_PENDING -DHAVE_RUBY_ENC_FIND_BASENAME -DPACKAGE_VERSION=\"8.6\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DHAVE_SYS_TIME_H=1 -DSTDC_HEADERS=1 -DHAVE_SYS_PARAM_H=1 -DUSE_THREAD_ALLOC=1 -D_REENTRANT=1 -D_THREAD_SAFE=1 -DHAVE_PTHREAD_ATTR_SETSTACKSIZE=1 -DHAVE_PTHREAD_ATFORK=1 -DTCL_THREADS=1 -DTCL_CFGVAL_ENCODING=\"iso8859-1\" -DHAVE_ZLIB=1 -DMODULE_SCOPE=extern\ __attribute__\(\(__visibility__\(\"hidden\"\)\)\) -DHAVE_HIDDEN=1 -DHAVE_CAST_TO_UNION=1 -DHAVE_VFORK=1 -DHAVE_POSIX_SPAWNP=1 -DHAVE_POSIX_SPAWN_FILE_ACTIONS_ADDDUP2=1 -DHAVE_POSIX_SPAWNATTR_SETFLAGS=1 -DTCL_SHLIB_EXT=\".so\" -DMP_PREC=4 -D_LARGEFILE64_SOURCE=1 -DTCL_WIDE_INT_IS_LONG=1 -DHAVE_GETCWD=1 -DHAVE_MKSTEMP=1 -DHAVE_OPENDIR=1 -DHAVE_STRTOL=1 -DHAVE_WAITPID=1 -DHAVE_GETNAMEINFO=1 -DHAVE_GETADDRINFO=1 -DHAVE_FREEADDRINFO=1 -DHAVE_GAI_STRERROR=1 -DHAVE_STRUCT_ADDRINFO=1 -DHAVE_STRUCT_IN6_ADDR=1 -DHAVE_STRUCT_SOCKADDR_IN6=1 -DHAVE_STRUCT_SOCKADDR_STORAGE=1 -DHAVE_GETPWUID_R_5=1 -DHAVE_GETPWUID_R=1 -DHAVE_GETPWNAM_R_5=1 -DHAVE_GETPWNAM_R=1 -DHAVE_GETGRGID_R_5=1 -DHAVE_GETGRGID_R=1 -DHAVE_GETGRNAM_R_5=1 -DHAVE_GETGRNAM_R=1 -DHAVE_DECL_GETHOSTBYNAME_R=1 -DHAVE_GETHOSTBYNAME_R_6=1 -DHAVE_GETHOSTBYNAME_R=1 -DHAVE_DECL_GETHOSTBYADDR_R=1 -DHAVE_GETHOSTBYADDR_R_8=1 -DHAVE_GETHOSTBYADDR_R=1 -DHAVE_TERMIOS_H=1 -DHAVE_SYS_IOCTL_H=1 -DTIME_WITH_SYS_TIME=1 -DHAVE_GMTIME_R=1 -DHAVE_LOCALTIME_R=1 -DHAVE_MKTIME=1 -DHAVE_TM_GMTOFF=1 -DHAVE_TIMEZONE_VAR=1 -DHAVE_STRUCT_STAT_ST_BLOCKS=1 -DHAVE_STRUCT_STAT_ST_BLKSIZE=1 -DHAVE_BLKCNT_T=1 -DHAVE_INTPTR_T=1 -DHAVE_UINTPTR_T=1 -DNO_UNION_WAIT=1 -DHAVE_SIGNED_CHAR=1 -DHAVE_LANGINFO=1 -DHAVE_MKSTEMPS=1 -DHAVE_FTS=1 -DTCL_UNLOAD_DLLS=1 -DUSE_DTRACE=1 -DHAVE_CPUID=1 -DNDEBUG=1 -DTCL_CFG_OPTIMIZED=1 -DHAVE_PW_GECOS=1 -DHAVE_LIBXFT=1 -DHAVE_XFT=1 -DHAVE_XKBKEYCODETOKEYSYM=1 -DXKEYCODETOKEYSYM_IS_DEPRECATED=1    -DWITH_TCL_ENABLE_THREAD=1 -fPIC -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Wno-complain-wrong-lang -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -mtls-dialect=gnu2 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNO_TAINT -m64 -o tcltklib.o -c tcltklib.c

tcltklib.c: In function ‘call_DoOneEvent’:
tcltklib.c:2045:21: error: passing argument 1 of ‘tk_funcall’ from incompatible pointer type [-Wincompatible-pointer-types]
 2045 |   return tk_funcall(call_DoOneEvent_core, 0, (VALUE*)NULL, flag_val);
      |                     ^~~~~~~~~~~~~~~~~~~~
      |                     |
      |                     VALUE (*)(VALUE) {aka long unsigned int (*)(long unsigned int)}

tcltklib.c:3382:37: error: passing argument 2 of ‘tcl_protect_core’ from incompatible pointer type [-Wincompatible-pointer-types]
 3382 |     code = tcl_protect_core(interp, proc, data);
      |                                     ^~~~
      |                                     |
      |                                     VALUE (*)(void) {aka long unsigned int (*)(void)}

tcltklib.c:7000:13: error: too many arguments to function ‘q->func’
 7000 |     ret = (q->func)(q->interp, q->argc, q->argv);
      |           ~~^~~~~~~

tcltklib.c:7088:18: error: too many arguments to function ‘func’
 7088 |         result = (func)(ip_obj, argc, argv);
      |                  ^
```

(And not erros but there are many warnings for using K&R style function declaration )
```
tcltklib.c:784:1: warning: old-style function definition [-Wold-style-definition]
  784 | get_ip(self)
      | ^~~~~~
```

--- a/ext/tk/tcltklib.c
+++ b/ext/tk/tcltklib.c
@@ -255,7 +255,7 @@ static ID ID_inspect;
 static VALUE ip_invoke_real _((int, VALUE*, VALUE));
 static VALUE ip_invoke _((int, VALUE*, VALUE));
 static VALUE ip_invoke_with_position _((int, VALUE*, VALUE, Tcl_QueuePosition));
-static VALUE tk_funcall _((VALUE(), int, VALUE*, VALUE));
+static VALUE tk_funcall _((VALUE(*)(VALUE, int, VALUE *), int, VALUE*, VALUE));
 
 /* Tcl's object type */
 #if TCL_MAJOR_VERSION >= 8
@@ -436,7 +436,7 @@ struct eval_queue {
 
 struct call_queue {
     Tcl_Event ev;
-    VALUE (*func)();
+    VALUE (*func)(VALUE, int, VALUE *);
     int argc;
     VALUE *argv;
     VALUE interp;
@@ -2017,12 +2017,7 @@ rbtk_EventCheckProc(ClientData clientData, int flag)
 
 #ifdef RUBY_USE_NATIVE_THREAD  /* Ruby 1.9+ !!! */
 static VALUE
-#ifdef HAVE_PROTOTYPES
-call_DoOneEvent_core(VALUE flag_val)
-#else
-call_DoOneEvent_core(flag_val)
-    VALUE flag_val;
-#endif
+call_DoOneEvent_core(VALUE flag_val, RB_UNUSED_VAR(int argc), RB_UNUSED_VAR(VALUE *argv))
 {
     int flag;
 
@@ -3365,7 +3360,7 @@ tcl_protect_core(interp, proc, data) /* should not raise exception */
 static int
 tcl_protect(interp, proc, data)
     Tcl_Interp *interp;
-    VALUE (*proc)();
+    VALUE (*proc)(VALUE);
     VALUE data;
 {
     int code;
@@ -7039,11 +7034,7 @@ call_queue_handler(evPtr, flags)
 }
 
 static VALUE
-tk_funcall(func, argc, argv, obj)
-    VALUE (*func)();
-    int argc;
-    VALUE *argv;
-    VALUE obj;
+tk_funcall(VALUE (*func)(VALUE, int, VALUE *), int argc, VALUE *argv, VALUE obj)
 {
     struct call_queue *callq;
     struct tcltkip *ptr;

